src/drivers/cmakeDriver.ts

```diff
@@ -1223,7 +1247,8 @@ export abstract class CMakeDriver implements vscode.Disposable {
         try {
             // _beforeConfigureOrBuild needs to refresh expansions early because it reads various settings
             // (example: cmake.sourceDirectory).
-            await this._refreshExpansions();
+            const configurePreset: preset.ConfigurePreset | undefined | null = (trigger === ConfigureTrigger.taskProvider) ? taskPreset : this._configurePreset;
+            await this._refreshExpansions(taskPreset);
```

Comment:
Should this be `configurePreset` instead of `taskPreset`?

Reply:
It actually has the same results, but I wanted pass "undefined"  for all the scenarios that doesn't have a `presetOverride` to keep the behavior as before. To avoid confusion I moved the `configurePreset` a few lines below where it is actually being used.